src/lib/mina_net2/bitswap_block.ml

```diff
@@ -0,0 +1,237 @@
+open Core_kernel
+
+let or_error_list_bind ls ~f =
+  let open Or_error.Let_syntax in
+  let rec loop = function
+    | [] ->
+        return []
+    | h :: t ->
+        let%bind r = f h in
+        let%map t' = loop t in
+        r :: t'
+  in
+  loop ls >>| List.concat
+
+type link = Blake2.t
+
+let link_size = Blake2.digest_size_in_bytes
+
+let absolute_max_links_per_block = Stdint.Uint16.(to_int max_int)
+
+type schema =
+  { total_blocks : int
+  ; full_link_blocks : int
+  ; non_max_link_block_count : int
+  ; last_block_data_size : int
+  ; max_block_size : int
+  ; max_links_per_block : int
+  }
+[@@deriving compare, eq, sexp]
+
+let required_bitswap_block_count ~max_block_size data_length =
```

Comment:
I added `requiredBitswapBlockCount` to the Go implementation. I'm not sure if I agree about making the names match 100%, just because the languages have different standards for how functions are named. OCaml tends to be more translation oriented `x_of_y y`, where are Go is more operation oriented `make_x(y)`. I was hoping to keep the algorithmic structure more or less the same while implementing the interface more idiomatically for OCaml.